cumulus_library/actions/uploader.py

```diff
@@ -71,7 +71,7 @@ def upload_files(args: dict):
     if args["target"]:
         filtered_paths = []
         for path in file_paths:
-            if any(study in str(path) for study in args["target"]):
+            if any(path.name.startswith(f"{study}__") for study in args["target"]):
```

Comment:
This is the actual code change here.

Before, this was a simple substring check. Which for my use case (`upload site/ -t core`) was hitting files like `site/data_metrics/data_metrics__count_c_us_core_v4_observation.parquet` because it has `core` in its filename, even though it's not a `core` study file.

So... I've changed this to check if the leaf filename starts with `study__` -- I could have additionally (or instead) checked that the directory above the file is just the study name? But this felt like a good first step.